src/rawtoaces_util/image_converter.cpp

```diff
@@ -826,80 +891,70 @@ bool ImageConverter::parse_parameters( const OIIO::ArgParse &arg_parser )
 
     std::string WB_method = arg_parser["wb-method"].get();
 
-    if ( WB_method == "metadata" )
-    {
-        settings.WB_method = Settings::WBMethod::Metadata;
-    }
-    else if ( WB_method == "illuminant" )
-    {
-        settings.WB_method = Settings::WBMethod::Illuminant;
-    }
-    else if ( WB_method == "box" )
-    {
-        settings.WB_method = Settings::WBMethod::Box;
-    }
-    else if ( WB_method == "custom" )
-    {
-        settings.WB_method = Settings::WBMethod::Custom;
-    }
-    else
+    static std::map<std::string, Settings::WBMethod> wb_methods = {
+        { "metadata", Settings::WBMethod::Metadata },
+        { "illuminant", Settings::WBMethod::Illuminant },
+        { "box", Settings::WBMethod::Box },
+        { "custom", Settings::WBMethod::Custom }
+    };
+
+    auto wb_it = wb_methods.find( WB_method );
+    if ( wb_it == wb_methods.end() )
     {
         std::cerr << std::endl
-                  << "Unsupported white balancing method: \"" << WB_method
-                  << "\"." << std::endl;
-
+                  << "ERROR: unsupported white balancing method '" << WB_method
+                  << "'. The following methods are supported: '"
+                  << OIIO::Strutil::join( get_map_keys( wb_methods ), "', '" )
+                  << "'." << std::endl;
         return false;
```

Comment:
Two improvements in one. No need for repeating code in if/else + listing available options in error.